src/main/java/pl/oskarkowalski/recruitment/ticketbookingapp/services/ReservationService.java

```diff
@@ -0,0 +1,171 @@
+package pl.oskarkowalski.recruitment.ticketbookingapp.services;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+import pl.oskarkowalski.recruitment.ticketbookingapp.dtos.ReservationDto;
+import pl.oskarkowalski.recruitment.ticketbookingapp.dtos.TicketDto;
+import pl.oskarkowalski.recruitment.ticketbookingapp.entities.Reservation;
+import pl.oskarkowalski.recruitment.ticketbookingapp.entities.Screening;
+import pl.oskarkowalski.recruitment.ticketbookingapp.entities.Seat;
+import pl.oskarkowalski.recruitment.ticketbookingapp.entities.Ticket;
+import pl.oskarkowalski.recruitment.ticketbookingapp.enums.SeatRow;
+import pl.oskarkowalski.recruitment.ticketbookingapp.enums.TicketPrice;
+import pl.oskarkowalski.recruitment.ticketbookingapp.exceptions.ResourceNotFoundException;
+import pl.oskarkowalski.recruitment.ticketbookingapp.repositories.ScreeningRepository;
+import pl.oskarkowalski.recruitment.ticketbookingapp.repositories.TicketRepository;
+
+import java.time.LocalDateTime;
+import java.util.Arrays;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+@Service
+public class ReservationService {
+
+  private final ScreeningRepository screeningRepository;
+  private final TicketRepository ticketRepository;
+
+  @Value("${discount.code}")
+  private char[] discountCode;
+
+  @Autowired
+  public ReservationService(ScreeningRepository screeningRepository, TicketRepository ticketRepository) {
+    this.screeningRepository = screeningRepository;
+    this.ticketRepository = ticketRepository;
+  }
+
+  public boolean isOkayToBuy(ReservationDto reservationDto) {
```

Comment:
`okay` to dobre słowo?

Reply:
Lepsze byłoby „Possible” lub „Valid”. Do poprawienia.